File: FloorballTraining.Web/Models/FloorballTrainingConfiguration/Training.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FloorballTraining.Web.Models.FloorballTrainingConfiguration
{
    [Table("Trainings", Schema = "dbo")]
    public partial class Training
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [ConcurrencyCheck]
        public string Name { get; set; }

        [ConcurrencyCheck]
        public string Description { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int Duration { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int PersonsMin { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int PersonsMax { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int Intensity { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int Difficulty { get; set; }

        [ConcurrencyCheck]
        public string CommentBefore { get; set; }

        [ConcurrencyCheck]
        public string CommentAfter { get; set; }

        [Required]
        [ConcurrencyCheck]
        public int PlaceId { get; set; }

        public Place Place { get; set; }

        [ConcurrencyCheck]
        public int? TrainingGoal1Id { get; set; }

        public Tag Tag { get; set; }

        [ConcurrencyCheck]
        public int? TrainingGoal2Id { get; set; }

        public Tag Tag1 { get; set; }

        [ConcurrencyCheck]
        public int? TrainingGoal3Id { get; set; }

        public Tag Tag2 { get; set; }

        [ConcurrencyCheck]
        public int GoaliesMax { get; set; }

        [ConcurrencyCheck]
        public int GoaliesMin { get; set; }

        public ICollection<Appointment> Appointments { get; set; }

        public ICollection<TrainingAgeGroup> TrainingAgeGroups { get; set; }

        public ICollection<TrainingPart> TrainingParts { get; set; }
    }
}

File: FloorballTraining.UI/src/app/models/Training.ts
import type {TrainingGoal} from "./TrainingGoal.ts";
import type {TrainingAgeGroup} from "./TrainingAgeGroup.ts";
import type {TrainingPart} from "./TrainingPart.ts";

export type Training = {
    name: string
    description: any
    duration: number
    personsMin: number
    personsMax: number
    goaliesMin: number
    goaliesMax: number
    intensity: number
    difficulty: number
    commentBefore: any
    commentAfter: any
    environment: number
    trainingGoal1: TrainingGoal
    trainingGoal3?: TrainingGoal
    trainingGoal2?: TrainingGoal
    trainingAgeGroups: TrainingAgeGroup[]
    trainingParts: TrainingPart[]
    id: number;
    
    // Vrací pole TrainingGoal, které obsahuje pouze goal1, goal2, goal3 pokud mají hodnotu
}


export function getTrainingGoals(training: Training): TrainingGoal[] {
    const goals: TrainingGoal[] = [];
    if (training.trainingGoal1) goals.push(training.trainingGoal1);
    if (training.trainingGoal2) goals.push(training.trainingGoal2);
    if (training.trainingGoal3) goals.push(training.trainingGoal3);
    return goals;
}
